File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/CreateLayerExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class CreateLayerExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("Authorization", "YOUR_API_KEY");

        var layerPost = new LayerPost(
            key: "checkout-flow",
            name: "Checkout Flow",
            description: "description_string"
        );

        try
        {
            var response = new LayersApi(config).CreateLayer(
                projectKey: "projectKey_string",
                layerPost: layerPost
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling LayersApi#CreateLayer: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/CreateLayerExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const layerPost: api.LayerPost = {
  key: "checkout-flow",
  name: "Checkout Flow",
  description: "description_string",
};

new api.LayersApi(configuration).createLayer(
  "projectKey_string", // projectKey
  layerPost,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling LayersApi#createLayer:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
